### PR TITLE
18469 Capitalize AGM in output titles

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#18469

*Description of changes:*
- app version = 7.0.11
- added missing cases for AGM filings
- added special case to convert "Agm" to "AGM"
- fixed a lint warning

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/services/enum-utilities.ts [hunk 1]
```diff
@@ -299,6 +299,8 @@ export default class EnumUtilities {
       case FilingTypes.ADMIN_FREEZE:
         // FUTURE: add freeze/unfreeze checks here
         return FilingNames.ADMIN_FREEZE
+      case FilingTypes.AGM_EXTENSION: return FilingNames.AGM_EXTENSION
+      case FilingTypes.AGM_LOCATION_CHANGE: return FilingNames.AGM_LOCATION_CHANGE
```
Comment (severinbeauvais) [inline src/services/enum-utilities.ts:303]: I think this method isn't currently used for the AGM filings, but it's better for this method to be complete anyway.

File: src/services/enum-utilities.ts [hunk 2]
```diff
@@ -332,13 +334,15 @@ export default class EnumUtilities {
...
+   * Converts a string in "camelCase" (or "PascalCase") to a string of separate, title-case words,
    * suitable for a title or proper name.
    * @param s the string to convert
    * @returns the converted string
    */
   static camelCaseToWords (s: string): string {
-    return s?.split(/(?=[A-Z])/).join(' ').replace(/^\w/, c => c.toUpperCase()) || ''
+    const words = s?.split(/(?=[A-Z])/).join(' ').replace(/^\w/, c => c.toUpperCase()) || ''
+    // SPECIAL CASE: convert 'Agm' to uppercase
+    return words.replace('Agm', 'AGM')
```
Comment (severinbeauvais) [inline src/services/enum-utilities.ts:345]: The alternative is a large effort to include the document name in the API response (requiring API work and possibly informing third parties/delays) and then use that.

This is a special case, but it's a simple one, and we could add more here if needed, so I think it's not the best solution but it's a solution we can afford and it's functional and understandable and safe. Thoughts?

**Comment by severinbeauvais** [comment]
Before:
image

After:
image

File: src/services/enum-utilities.ts (hunk 2 above)
Comment (JazzarKarim) [reply src/services/enum-utilities.ts:345]: I think it's OK for now Sev. If more and more arise in the future, then we can think about going the long and hard way.

**Review by JazzarKarim — APPROVED** [review]
LGTM!

**Review by ketaki-deodhar — APPROVED** [review]
Solution without big changes for now. this works well. LGTM!

**Pull request merged**